querydsl-core/src/main/java/com/querydsl/core/DefaultQueryMetadata.java

```diff
@@ -306,6 +301,9 @@ public void setLimit(Long limit) {
 
     @Override
     public void setModifiers(@Nullable QueryModifiers restriction) {
+        if (restriction == null) {
+            throw new NullPointerException();
```

Comment:
I removed the `@Nullable` annotation from the argument. I think getters/setters should be mostly side-effect free, that's why  I'd rather not do a silent conversion. This way it is maybe more clear what the valid values for this property are.